// AUTOMATICALLY GENERATED, DO NOT EDIT!

syntax = "proto3";

package NYP.NServer.NObjects.NProto;


import "yp/client/api/proto/data_model.proto";
import "yp/client/api/proto/replica_set.proto";
import "yp/client/api/proto/multi_cluster_replica_set.proto";
import "yp/client/api/proto/resource_cache.proto";
import "yp/client/api/proto/dynamic_resource.proto";
import "yp/client/api/proto/stage.proto";
import "yp/client/api/proto/host_infra.proto";
import "yp/client/api/proto/deploy.proto";
import "yt/core/misc/proto/error.proto";
import "yt/core/ytree/proto/attributes.proto";
import "yt/core/yson/proto/protobuf_interop.proto";
import "yp/client/api/proto/enums.proto";
import "yp/client/api/proto/conditions.proto";


////////////////////////////////////////////////////////////////////////////////

message TNodeStatusEtc
{
    uint64 last_handshake_time = 10;
    string last_handshake_master_address = 11;
    string agent_version = 8;
    NYP.NClient.NApi.NProto.TNodeStatus.THfsm hfsm = 5;
    NYP.NClient.NApi.NProto.TNodeStatus.TMaintenance maintenance = 6;
    repeated string unknown_pod_ids = 7;
}

////////////////////////////////////////////////////////////////////////////////

message TPodSpecEtc
{
    NYP.NClient.NApi.NProto.TPodSpec.TResourceRequests resource_requests = 3;
    repeated NYP.NClient.NApi.NProto.TPodSpec.TIP6AddressRequest ip6_address_requests = 5;
    repeated NYP.NClient.NApi.NProto.TPodSpec.TIP6SubnetRequest ip6_subnet_requests = 6;
    repeated NYP.NClient.NApi.NProto.TPodSpec.TDiskVolumeRequest disk_volume_requests = 13;
    repeated NYP.NClient.NApi.NProto.TPodSpec.TGpuRequest gpu_requests = 23;
    string node_filter = 8;
    repeated NYP.NClient.NApi.NProto.TPodSpec.THostDevice host_devices = 10;
    repeated NYP.NClient.NApi.NProto.TPodSpec.TSysctlProperty sysctl_properties = 11;
    NYP.NClient.NApi.NProto.TPodSpec.TVirtualServiceOptions virtual_service_options = 12;
    NYP.NClient.NApi.NProto.EPodHostNameKind host_name_kind = 15;
    NYP.NClient.NApi.NProto.EPodOutOfMemoryPolicy out_of_memory_policy = 21;
    NYP.NClient.NApi.NProto.THostInfraInfo host_infra = 22;
}

////////////////////////////////////////////////////////////////////////////////

message TPodStatusEtc
{
    repeated NYP.NClient.NApi.NProto.TPodStatus.TResourceAllocation scheduled_resource_allocations = 4;
    repeated NYP.NClient.NApi.NProto.TPodStatus.TIP6AddressAllocation ip6_address_allocations = 5;
    repeated NYP.NClient.NApi.NProto.TPodStatus.TIP6SubnetAllocation ip6_subnet_allocations = 6;
    repeated NYP.NClient.NApi.NProto.TPodStatus.TDiskVolumeAllocation disk_volume_allocations = 14;
    repeated NYP.NClient.NApi.NProto.TPodStatus.TGpuAllocation gpu_allocations = 16;
    NYP.NClient.NApi.NProto.TPodStatus.TDns dns = 7;
    NYP.NClient.NApi.NProto.TPodStatus.TEviction eviction = 8;
}

////////////////////////////////////////////////////////////////////////////////

message TPodAgentStatusEtc
{
    NYT.NProto.TError execution_error = 6;
    NYT.NProto.TError install_error = 9;
    uint64 failed_install_attempt_spec_timestamp = 10;
    repeated string validation_failures = 8;
    uint64 last_heartbeat_time = 7;
    uint64 current_spec_timestamp = 11;
    NYP.NClient.NApi.NProto.TCondition current_spec_applied = 12;
}

////////////////////////////////////////////////////////////////////////////////

message TPodStatusSchedulingEtc
{
    NYP.NClient.NApi.NProto.ESchedulingState state = 1;
    uint64 last_updated = 3;
    string message = 4;
    NYT.NProto.TError error = 5;
}

////////////////////////////////////////////////////////////////////////////////

message TAccountSpecEtc
{
    NYP.NClient.NApi.NProto.TResourceTotals resource_limits = 2;
}

////////////////////////////////////////////////////////////////////////////////

message TReplicaSetSpecEtc
{
    string revision_id = 1;
    uint32 replica_count = 2;
    NYP.NClient.NApi.NProto.TPodTemplateSpec pod_template_spec = 3;
    NYP.NClient.NApi.NProto.TReplicaSetSpec.TDeploymentStrategy deployment_strategy = 4;
    NYP.NClient.NApi.NProto.TReplicaSetSpec.TConstraints constraints = 6;
}

////////////////////////////////////////////////////////////////////////////////

message TMultiClusterReplicaSetSpecEtc
{
    uint64 revision = 6;
    NYP.NClient.NApi.NProto.TPodTemplateSpec pod_template_spec = 3;
    NYP.NClient.NApi.NProto.TMultiClusterReplicaSetSpec.TDeploymentStrategy deployment_strategy = 4;
    repeated NYP.NClient.NApi.NProto.TMultiClusterReplicaSetSpec.TClusterReplicaSetSpecPreferences clusters = 5;
}

////////////////////////////////////////////////////////////////////////////////

message TStageSpecEtc
{
    uint64 revision = 1;
    map<string, NYP.NClient.NApi.NProto.TDeployUnitSpec> deploy_units = 2 [(NYT.NYson.NProto.yson_map) = true];
    NYP.NClient.NApi.NProto.TStageSpec.TRevisionInfo revision_info = 4;
}

////////////////////////////////////////////////////////////////////////////////
